### DISCUSSION TITLE
How do i 'use' a trained model?

### DISCUSSION THREAD

**Discussion by vegovs** [Lightning-AI/pytorch-lightning#7200] [code help: CV]
I am trying to use a model trained with Pytorch Lightning via torch.
As far as I have googled, this is the way:

```
model = torch.load(path_to_model_checkpoint)
model.eval()
segmented_buildings = model(image)
```

However, the first error I got when doing this was that Pytorch Lightning module was missing.
Did not know it was needed when just running inference with the weights via torch? 
Anyway, after installing the module I got a new error:

```
AttributeError: Can't get attribute '_gpus_arg_default' on <module 'pytorch_lightning.utilities.argparse' from '/home/user/venvs/orthophoto-dataset-generator/lib/python3.8/site-packages/pytorch_lightning/utilities/argparse.py'>
```

What am I doing wrong? Or is this a bug?

**Comment by awaelchli** [comment]
This is a bug we recently had to fix. You need to skip Lightning version 1.2.8 and either install <= 1.2.7 or  >= 1.3.0rc1

**Comment by awaelchli** [comment]
```python
model = YourModelCls.load_from_checkpoint(path_to_model_checkpoint)
model.eval()
segmented_buildings = model(image)
```
